pkg/controllers/applicationset_controller_test.go

```diff
@@ -1732,6 +1752,16 @@ func TestValidateGeneratedApplications(t *testing.T) {
 					assert.True(t, foundMatch, "Unble to locate expected error: %s", cc.expectedErrors)
 					matched = matched || foundMatch
 				}
+				// validation message was returned: it should be expected
+				matched = false
```

Comment:
I think this `matched = false` should be removed, otherwise the `match` logic above this line is ignored (correct me if I am wrong)

Reply:
I guess I missed out an assert statement before this line. I have added that. The above `matched=false` will reset the value for matched for test in the section below it.